syntax = "proto3";

package protobuf;
message TestDataClass {
  optional bool bool1 = 1;
  optional bool bool2 = 2;
  // optional byte1 = 3; // todo: data types missing in protobuf
  // optional byte2 = 4;
  // optional char1 = 5;
  // optional char2 = 6;
  // optional short1 = 7;
  // optional short2 = 8;
  optional int32 int1 = 9;
  optional int32 int2 = 10;
  optional int64 long1 = 11;
  optional int64 long2 = 12;
  optional float float1 = 13;
  optional float float2 = 14;
  optional double double1 = 15;
  optional double double2 = 16;
  optional string string1 = 17;
  optional string string2 = 18;
  // arrays
  repeated bool boolArray = 19;
  // repeated byteArray = 20;
  // repeated shortArray = 21;
  repeated int32 intArray = 22;
  repeated int64 longArray = 23;
  repeated float floatArray = 24;
  repeated double doubleArray = 25;
  repeated string stringArray = 26;
  // multi dim data, flattened
  repeated int32 nDimensions = 27;
  repeated bool boolNdimArray = 28;
  // repeated byteNdimArray = 29;
  //repeated shortNdimArray = 30;
  repeated int32 intNdimArray = 31;
  repeated int64 longNdimArray = 32;
  repeated float floatNdimArray = 33;
  repeated double doubleNdimArray = 34;
  optional TestDataClass nestedData = 35;
}